src/devices/Bh1745/Bh1745.cs

```diff
@@ -0,0 +1,394 @@
+                _higherInterruptThreshold = value;
+            }
+        }
+
+        /// <summary>
+        /// Gets or sets the channel compensation multipliers which are used to compensate the measurements.
+        /// </summary>
+        public ChannelCompensationMultipliers ChannelCompensationMultipliers { get; set; }
+
+        /// <summary>
+        /// Resets the device to the default configuration.
+        /// On reset the sensor goes to power down mode.
+        /// </summary>
+        public void Reset()
+        {
+            var status = Read8BitsFromRegister((byte)Register.SYSTEM_CONTROL);
+            status = (byte)((status & (byte)~Mask.SW_RESET) | 0x01 << 7);
+
+            Write8BitsToRegister((byte)Register.SYSTEM_CONTROL, status);
+
+            // set default measurement configuration
+            MeasurementTime = MeasurementTime.Ms160;
+            AdcGain = AdcGain.X1;
+            MeasurementIsActive = true;
+
+            // write default value to Mode_Control3
+            Write8BitsToRegister((byte)Register.MODE_CONTROL3, 0x02);
+        }
+
+        /// <summary>
+        /// Reads whether the last measurement is valid.
+        /// </summary>
+        public bool ReadMeasurementIsValid()
+        {
+            var valid = Read8BitsFromRegister((byte)Register.MODE_CONTROL2);
+            valid = (byte)(valid & (byte)Mask.VALID);
+            return Convert.ToBoolean(valid);
+        }
+
+        /// <summary>
+        /// Reads the red data register of the sensor.
+        /// </summary>
+        /// <returns></returns>
+        public ushort ReadRedDataRegister() => Read16BitsFromRegister((byte)Register.RED_DATA);
+
+        /// <summary>
+        /// Reads the green data register of the sensor.
+        /// </summary>
+        /// <returns></returns>
+        public ushort ReadGreenDataRegister() => Read16BitsFromRegister((byte)Register.GREEN_DATA);
+
+        /// <summary>
+        /// Reads the blue data register of the sensor.
+        /// </summary>
+        /// <returns></returns>
+        public ushort ReadBlueDataRegister() => Read16BitsFromRegister((byte)Register.BLUE_DATA);
+
+        /// <summary>
+        /// Reads the clear data register of the sensor.
+        /// </summary>
+        /// <returns></returns>
+        public ushort ReadClearDataRegister() => Read16BitsFromRegister((byte)Register.CLEAR_DATA);
+
+        /// <summary>
+        /// Gets the compensated color reading from the sensor.
+        /// </summary>
+        /// <returns></returns>
+        public Color GetCompensatedColor()
+        {
+            var clearDataRaw = ReadClearDataRegister();
+            if (clearDataRaw == 0)
+                return Color.FromArgb(0, 0, 0);
+
+            // apply channel multipliers and normalize
+            var compensatedRed = ReadRedDataRegister() * ChannelCompensationMultipliers.Red / MeasurementTime.ToMilliseconds() * 360;
```

Comment:
please use explicit types here (especially it seems it's same number of characters)